proto-router/src/main/kotlin/actor/proto/router/ConsistentHashRouterState.kt

```diff
@@ -5,22 +5,25 @@ import actor.proto.PID
 import actor.proto.send
 import actor.proto.toShortString
 
+class ConsistentHashRouterRoutees (val hashRing: HashRing,val routeeMap:Map<String, PID>)
+
 internal class ConsistentHashRouterState(private val hash: (String) -> Int, private var replicaCount: Int) : RouterState() {
-    private lateinit var hashRing: HashRing
-    private lateinit var routeeMap: Map<String, PID>
+    private lateinit var routees : ConsistentHashRouterRoutees
     override fun getRoutees(): Set<PID> {
-        return routeeMap.values.toSet()
+        return routees.routeeMap.values.toSet()
     }
 
     override fun setRoutees(routees: Set<PID>) {
-        routeeMap = mapOf()
-        var nodes: Set<String> = setOf()
+        var routeeMap = mapOf<String,PID>()
+        var nodes = setOf<String>()
         for (pid in routees) {
             val nodeName: String = pid.toShortString()
             nodes += nodeName
             routeeMap += Pair(nodeName, pid)
         }
-        hashRing = HashRing(nodes, hash, replicaCount)
+        val hashRing = HashRing(nodes, hash, replicaCount)
+
+       this.routees  = ConsistentHashRouterRoutees(hashRing,routeeMap)
```

Comment:
re-assign the routees field only once everything is recalculated and added to the ConsistentHashRouterRoutees container